bigbone/src/main/kotlin/social/bigbone/api/entity/Suggestion.kt

```diff
@@ -0,0 +1,46 @@
+package social.bigbone.api.entity
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents a suggested account to follow and an associated reason for the suggestion.
+ * @see <a href="https://docs.joinmastodon.org/entities/Suggestion/">Mastodon API Suggestion</a>
+ */
+
+@Serializable
+data class Suggestion(
+
+    /**
+     * The reason this account is being suggested.
+     */
+    @SerialName("source")
+    val source: String = SourceSuggestion.STAFF.value,
+
+    /**
+     * The account being recommended to follow.
+     */
+    @SerialName("account")
+    val account: Account
+) {
+
+    /**
+     * Represents a suggested account to follow and an associated reason for the suggestion.
+     */
+    enum class SourceSuggestion(val value: String) {
+        /**
+         * This account was manually recommended by your administration team.
+         */
+        STAFF("staff"),
+
+        /**
+         * You have interacted with this account previously.
+         */
+        PAST_INTERACTIONS("past_interactions"),
+
+        /**
+         * This account has many reblogs, favourites, and active local followers within the last 30 days.
+         */
+        GLOBAL("global")
+    }
```

Comment:
If we return the `SourceSuggestion` instead of its string value, we can simplify this:

_(please mind the additional `@Serializable` and `@SerialName` attributes!)_
```suggestion
    @Serializable
    enum class SourceSuggestion {
        /**
         * This account was manually recommended by your administration team.
         */
        @SerialName("staff")
        STAFF,

        /**
         * You have interacted with this account previously.
         */
        @SerialName("past_interactions")
        PAST_INTERACTIONS,

        /**
         * This account has many reblogs, favourites, and active local followers within the last 30 days.
         */
        @SerialName("global")
        GLOBAL
    }
```

Reply:
I like this simplification. In the project, we have lots of other enum classes which also could be simplified. Would be cool, if we can simplifiy those as well. Great starter task. Let's create a separate issue for this.

Reply:
New issue created: https://github.com/andregasser/bigbone/issues/305